### ISSUE TITLE
Debug physics breakdown at specific location

### ISSUE THREAD

**Issue by pravusjif** [decentraland/unity-renderer#494]
If I [spawn at -11, -11](https://play.decentraland.zone/index.html?renderer=urn:decentraland:off-chain:renderer-artifacts:feat/generic_metrics_controller&ENV=org&position=-11,-11) and move a bit to wait for the surrounding scenes to finish loading, something breaks the character collisions and I can for through any collider... somehow this breaks other stuff as the NFTShapes in the Genesis Plaza gallery are not there when this bug is triggered.

Doing commit hunting I couldn't find when this was introduced (tried going way back and I can still reproduce it) so I'm guessing one of the scenes at that location is breaking the explorer...

![Screen Shot 2021-05-24 at 14.10.35.png](https://images.zenhubusercontent.com/5d9200f4f58e25000108abac/a33c01ec-313f-4f26-a994-988abd6ea4b4)

```
Invalid worldAABB. Object is too large or too far away from the origin.

Invalid parameter pos in Collider.FetchPoseFromTransform because it was infinity or NaN
UnityEngine.StackTraceUtility:ExtractStackTrace ()
DCL.PhysicsSyncController:Sync () (at Assets/Scripts/MainScripts/DCL/Controllers/PhysicsSyncController/PhysicsSyncController.cs:32)
DCLCharacterController:LateUpdate () (at Assets/Scripts/MainScripts/DCL/Controllers/CharacterController/DCLCharacterController.cs:348)

Invalid AABB rkBox
UnityEngine.Bounds:ClosestPoint (UnityEngine.Vector3)
DCL.Rendering.CullingController/<ProcessProfile>d__26:MoveNext () (at Assets/Rendering/Culling/CullingController.cs:178)
UnityEngine.SetupCoroutine:InvokeMoveNext (System.Collections.IEnumerator,intptr)

Invalid AABB *this
UnityEngine.Bounds:Contains (UnityEngine.Vector3)
DCL.Rendering.CullingController/<ProcessProfile>d__26:MoveNext () (at Assets/Rendering/Culling/CullingController.cs:180)
UnityEngine.SetupCoroutine:InvokeMoveNext (System.Collections.IEnumerator,intptr)

Invalid AABB a

Assertion failed on expression: 'IsFinite(distanceForSort)'

Assertion failed on expression: 'IsFinite(distanceAlongView)'

Invalid worldAABB. Object is too large or too far away from the origin.
UnityEngine.Renderer:get_bounds ()
DCL.Helpers.Utils:GetSafeBounds (UnityEngine.Renderer) (at Assets/Scripts/MainScripts/DCL/Helpers/Utils/Utils.cs:580)
DCL.Models.MeshesInfoUtils:BuildMergedBounds (UnityEngine.Renderer[]) (at Assets/Scripts/MainScripts/DCL/Controllers/Scene/MeshesInfo/MeshesInfoUtils.cs:18)
DCL.Models.MeshesInfo:RecalculateBounds () (at Assets/Scripts/MainScripts/DCL/Controllers/Scene/MeshesInfo/MeshesInfo.cs:89)
DCL.Models.MeshesInfo:get_mergedBounds () (at Assets/Scripts/MainScripts/DCL/Controllers/Scene/MeshesInfo/MeshesInfo.cs:53)
DCL.Controllers.SceneBoundsChecker:IsEntityInsideSceneBoundaries (DCL.Models.IDCLEntity) (at Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneBoundariesController/SceneBoundsChecker.cs:183)
DCL.Controllers.SceneBoundsChecker:EvaluateMeshBounds (DCL.Models.IDCLEntity) (at Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneBoundariesController/SceneBoundsChecker.cs:200)
DCL.Controllers.SceneBoundsChecker:EvaluateEntityPosition (DCL.Models.IDCLEntity) (at Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneBoundariesController/SceneBoundsChecker.cs:178)
DCL.Controllers.SceneBoundsChecker:EvaluateEntityPosition (DCL.Models.IDCLEntity) (at Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneBoundariesController/SceneBoundsChecker.cs:157)
DCL.Controllers.SceneBoundsChecker:EvaluateEntityPosition (DCL.Models.IDCLEntity) (at Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneBoundariesController/SceneBoundsChecker.cs:157)
DCL.Controllers.SceneBoundsChecker/<CheckEntities>d__21:MoveNext () (at Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneBoundariesController/SceneBoundsChecker.cs:64)
UnityEngine.SetupCoroutine:InvokeMoveNext (System.Collections.IEnumerator,intptr)
```

https://images.zenhubusercontent.com/337227404/50991e27-57e0-44ec-a104-1a4a3ef4a093/2.mov

https://images.zenhubusercontent.com/337227404/6d203b3c-20d4-4102-98e7-ad347f6066d6/3.mov

**Comment by pravusjif** [comment]
Upon further debugging, the problem is in GENESIS PLAZA, apparently when the clouds start moving, something f*ck everything up:

VID 1:
https://drive.google.com/file/d/18AdCqrKHAL7Qc9f3GXz7MPSF1hsP5J-5/view?usp=sharing

VID 2:
https://drive.google.com/file/d/134CeNxVF80kO8V4txH6RzQmXcJXpJWrF/view?usp=sharing

**Comment by pravusjif** [comment]
![Screen Shot 2021-05-25 at 21.17.32.png](https://images.zenhubusercontent.com/5d9200f4f58e25000108abac/46fe5484-d3c1-456f-a928-c12eb7186fb0)

Upon further experimentation, If I intercept the transform update mesage for entities being positioned super far away (pos magnitude 10k, 40k or more) I can detect the culprit entities.

Also, if in those cases we escape and avoid applying position + rotation + scale to the entity gameobject, **sometimes** the whole physics breakdown doesn't happen.

I'll keep investigating.

**Comment by pravusjif** [comment]
I concluded that what's happening is:
1. The Genesis Plaza scene breaks due to unknown circumstances when loading the scene from far away
2. The scene code crash somehow updates the position and scale of certain entities with very large values
3. The enormous values in position and scale end up breaking Unity's physics sync

I implemented [a PR](https://github.com/decentraland/unity-renderer/pull/506) with a fix to avoid the physics breakdown when messages with these kind of values appear.
